optuna/study/_optimize.py

```diff
@@ -239,6 +239,8 @@ def _log_failed_trial(
     trial: FrozenTrial, message: Union[str, Warning], exc_info: Any = None
 ) -> None:
     _logger.warning(
-        "Trial {} failed because of the following error: {}".format(trial.number, message),
+        "Trial {} failed because of the following error: {} and following parameters: {}.".format(
+            trial.number, message, trial.params
+        ),
```

Comment:
[nits] How about updating messages like the following?

```suggestion
        "Trial {} failed with parameters: {} because of the following error: {}.".format(
            trial.number, trial.params, message
        ),
```